src/main/java/io/github/thebusybiscuit/slimefun4/implementation/guide/SurvivalSlimefunGuide.java

```diff
@@ -541,10 +542,29 @@ private void displayItem(ChestMenu menu, PlayerProfile profile, Player p, Object
             return false;
         };
 
+        MenuClickHandler lockedClickHandler = (pl, slot, itemstack, action) -> {
+            try {
+                SlimefunItem sfitem = SlimefunItem.getByItem(itemstack);
+                this.unlockItem(pl, sfitem, player -> displayItem(profile, itemstack, 0, true));
+            } catch (Exception | LinkageError x) {
+                printErrorMessage(pl, x);
+            }
+            return false;
+        };
+
         boolean isSlimefunRecipe = item instanceof SlimefunItem;
 
         for (int i = 0; i < 9; i++) {
             ItemStack recipeItem = getDisplayItem(p, isSlimefunRecipe, recipe[i]);
+
+            if (recipeItem != null && recipeItem.hasItemMeta()) {
+                ItemMeta im = recipeItem.getItemMeta();
+                if (im.hasLore() && im.getLore().contains(ChatColor.DARK_RED + ChatColor.BOLD.toString() + SlimefunPlugin.getLocalization().getMessage(p, "guide.locked"))) {
+                    menu.addItem(recipeSlots[i], recipeItem, lockedClickHandler);
+                    continue;
+                }
+            }
```

Comment:
Still not a fan of this lore comparison here.

Something much more elegant would be to utilise PersistentData or even simpler, a new subclass of `ItemStack` called `UnlockableItemStack`.
Then all you'd need to do is to make `getDisplayItem` return this new class instead and do an `instanceof check` here.
Or perhaps the instanceof check could even happen inside of the click handler itself.

I think this would make this all look a lot cleaner and improve maintainability in the future.